### PR TITLE
Substitute custom utils with lodash

### PR DESCRIPTION
**Pull request by zcei**
Umbrella: #1333 

Please note that the tests are failing because the semantics of the merge, extend & forEach lodash methods are different to the custom ones.
Those will be fixed when the consumers are touched to use lodash directly instead of through `utils`.

### CODE REVIEW

**Comment by montogeek** [comment]
How much this change decreased or increased the bundle size?

**Comment by zcei** [comment]
The overall bundle size at the end in `release/1.0.0` (not necessarily in this PR) was still meeting the defined threshold of `bundlesize`. If I recall correctly, it was 4.95 kB.
The advantage is that every single project which is using `axios` & `lodash` at the same time (which has a somewhat high probability) can share their dependencies.

So for the end-user it is likely to result in a lower bundle size than before.

**Comment by Khaledgarbaya** [comment]
@montogeek we use tree shaking and the lodash webpack plugin so the bundle size will not be big.
Right now it's around `7.8kb` which will try to decrease along the way

File: lib/utils.js [hunk 1]
```diff
@@ -1,34 +1,25 @@
-'use strict'
-
-const bind = require('./helpers/bind')
-const isBuffer = require('is-buffer')
+import {
```
Comment (hans-lizihan) [inline lib/utils.js]: This will completely break npm run test:unit

Should we use require back?

File: lib/utils.js [hunk 2]
```diff
@@ -188,116 +109,25 @@ function isStandardBrowserEnv () {
...
- * If `obj` is an Array callback will be called passing
- * the value, index, and complete array for each item.
- *
- * If 'obj' is an Object callback will be called passing
- * the value, key, and complete object for each property.
- *
- * @param {Object|Array} obj The object to iterate
- * @param {Function} fn The callback to invoke for each item
- */
-function forEach (obj, fn) {
```
Comment (hans-lizihan) [inline lib/utils.js]: https://lodash.com/docs/4.17.4#forEach

if keys with `length` lodash forEach has different behavior with this custom forEach method

File: lib/utils.js (hunk 1 above)
Comment (axe312ger) [reply lib/utils.js]: AFAIK will we update the tests later on to be able to handle import.

Comment (hans-lizihan) [reply lib/utils.js]: got it. i have fixed it in https://github.com/axios/axios/pull/1407

File: lib/utils.js (hunk 2 above)
Comment (zcei) [reply lib/utils.js]: Thanks, that's part of the upcoming PRs described in #1333, where we update the usage of `forEach` to `forOwn` on two or three occasions that cause this trouble.

The problem here was that we implemented all these changes and only then sat down an split it to logical units, so some small changes might have landed in a different step than anticipated.

**Pull request merged**